cmd/registry/registryList.go

```diff
@@ -122,6 +141,48 @@ func printNodes(nodes []*k3d.Node, headersOff bool) {
 	})
 
 	for _, node := range nodes {
-		fmt.Fprintf(tabwriter, "%s\t%s\t%s\n", strings.TrimPrefix(node.Name, "/"), string(node.Role), node.Labels[k3d.LabelClusterName])
+		// try top find the port in the host that leads to port 5000 in the container
+		registryPortBinding, ok := node.Ports[registryPort]
+		if !ok {
+			log.Fatalf("Registry %s does not have any port binding for port %s: %+v",
+				node.Name, registryPort, node.Ports)
+		}
+		hostRegistryPort := registryPortBinding[0].HostPort
+		hostRegistryPortI, err := strconv.Atoi(hostRegistryPort)
+		if err != nil {
+			log.Fatalf("Could not parse registry port %s: %s", hostRegistryPort, err)
+		}
+
+		if outputFormat == "json" || outputFormat == "yaml" {
+			entry := jsonOutput{
+				Node: *node,
+				Port: hostRegistryPortI,
+			}
+
+			jsonOutputEntries = append(jsonOutputEntries, entry)
+		} else {
+			// TODO: maybe we should print the address:port users can use for pushing/pulling (ie, "registry.localhost:46333")
+			fmt.Fprintf(tabwriter, "%s\t%d\t%s\t%s\n",
+				strings.TrimPrefix(node.Name, "/"),
+				hostRegistryPortI,
+				string(node.Role),
+				node.Labels[k3d.LabelClusterName])
```

Comment:
This label is only set, if the registry was created together with a cluster via `k3d cluster create --registry-create`, but not, if the registry was created standalone, e.g. via `k3d registry create`.
May lead to an exception.

Reply:
That error was not introduced by this PR, right? I mean, I can fix it anyway, but just to be sure I've not introduced something broken...

Reply:
Unfortunately, I cannot see the change anymore, but maybe you just indented it and that's why it ended up in the changes here. I didn't spend much time on the `registry list` function, so it may very well be, that it was there before :+1: